### DISCUSSION TITLE
Is it possible to enable treesitter-context in the previewer?

### DISCUSSION THREAD

**Discussion by lfantonioli** [ibhagwan/fzf-lua#1509] [Q&A] [answered]
is it possible to enable something like [treesitter-context](https://github.com/nvim-treesitter/nvim-treesitter-context) in the previewer window (as a custom previewer)? 

I know running treesitter-context on all previewers might be a heavy operation, but for some operations, like lsp_references, the context would help to understand quickly your location in the code.

**Comment by ibhagwan** [comment]
Not possible ATM, the sitter context is an additional plugin dependency and IMHO unnecessary complexity to the previewer.

**Reply by lfantonioli** [reply]
Thank you for the confirmation. I also believe that such a feature should not be included in the plugin by default. I was just wondering if it would be possible to create a previewer, similar to the built-in previewer, but with specific plugins enabled.

**Reply by ibhagwan** [reply]
> Thank you for the confirmation. I also believe that such a feature should not be included in the plugin by default. I was just wondering if it would be possible to create a previewer, similar to the built-in previewer, but with specific plugins enabled.

Sure you can, take a look at the advanced Wiki builtin previewer section you can override a class function and populate it with whatever you want and probably enable a plugin too.

**Comment by ibhagwan** [comment]
@lfantonioli, recently a `multiwindow` option was added in `nvim-treesitter-context` which peeked my interest as this might turn possible, feel free to follow https://github.com/nvim-treesitter/nvim-treesitter-context/issues/534

**Reply by ibhagwan** [reply]
Unfortunately, [Looks like it won’t happen](https://github.com/nvim-treesitter/nvim-treesitter-context/issues/534#issuecomment-2517701892).

**Reply by lfantonioli** [reply]
Indeed, it's unfortunate that it won't be possible. But I really appreciate your effort in trying to make it happen. Thank you!

**Reply by ibhagwan** [reply]
> Indeed, it's unfortunate that it won't be possible. But I really appreciate your effort in trying to make it happen. Thank you!

On another note,  you might be interested in the latest integration with `render-markdown.nvim`  looks absolutely magical, see https://github.com/ibhagwan/fzf-lua/discussions/1546

**Reply by ibhagwan** [reply]
@lfantonioli , perhaps some progress, #1552

**Reply by ibhagwan** [reply, marked as answer]
@lfantonioli, FYI, with the latest commit add this to your `setup` call, make sure to update `nvim-treesitter-context`  as the `multiwindow` feature was only recently added and it won't work without it:
```lua
require("fzf-lua").setup({
  previewers = { builtin = { treesitter = { context = true} } }
})
```

Can also be called indivually:
```vim
:FzfLua live_grep previewer.treesitter.context=true
```

**Reply by lfantonioli** [reply]
Thank you so much, @ibhagwan, for making it happen. The treesitter-context integration is not only the coolest I’ve seen in a long time but also incredibly impressive in terms of the rapid progress and improvements you've delivered to the fzf-lua plugin.

**Reply by ibhagwan** [reply]
> Thank you so much, @ibhagwan, for making it happen. The treesitter-context integration is not only the coolest I’ve seen in a long time but also incredibly impressive in terms of the rapid progress and improvements you've delivered to the fzf-lua plugin.

Ty for the kind words and suggestion, I also find the context very useful.

**Reply by ibhagwan** [reply]
FYI, https://github.com/ibhagwan/fzf-lua/commit/c40666b848b88c9dc91a1e8b6e7a4fcf8db6e29d added `<F7>` as a default bind (customizable) to toggle (hide/unhide) the treesitter context.
